config/sources/families/include/rockchip64_common.inc

```diff
@@ -299,10 +299,42 @@ write_uboot_platform() {
 
 # @TODO: this is not ready for BOOT_SCENARIO=binman yet
 write_uboot_platform_mtd() {
-	if [[ -f $1/rkspi_loader.img ]]; then
-		dd if=$1/rkspi_loader.img of=$2 conv=notrunc status=none > /dev/null 2>&1
+	FILES=$(find "$1" -maxdepth 1 -type f -name "rkspi_loader*.img")
+	if [ -z "$FILES" ]; then
+		echo "No SPI image found."
+		exit 1
+	fi
+
+	MENU_ITEMS=()
+	i=1
+
+	# Read the files into an array
+	while IFS= read -r file; do
+		filename=$(basename "$file")
+		MENU_ITEMS+=("$i" "$filename" "")
+		((i++))
+	done <<< "$FILES"
+
+	# If there is only one image, we can skip the dialog
```

Comment:
Also skip and log if

- terminal not interactive (eg `[[ -t 0 ]]` for stdin if I am not mistaken)
- `dialog` not available